Mathlib/Analysis/Analytic/Constructions.lean

```diff
@@ -137,6 +137,87 @@ theorem AnalyticOn.curry_right {f : E × F → G} {x : E} {s : Set (E × F)} (fa
   fun y m ↦ (fa (x, y) m).along_snd
 alias AnalyticOn.along_snd := AnalyticOn.curry_right
 
+/-!
+### Analyticity in Pi spaces
+-/
+
+section
+
+variable {ι : Type*} [Fintype ι] {e : E} {Fm : ι → Type*}
+    [∀ i, NormedAddCommGroup (Fm i)] [∀ i, NormedSpace 𝕜 (Fm i)]
+    {f : Π i, E → Fm i} {s : Set E} {r : ℝ≥0∞}
+    {p : Π i, FormalMultilinearSeries 𝕜 E (Fm i)}
+
+lemma FormalMultilinearSeries.le_pi_radius (h : ∀ i, r ≤ (p i).radius) :
+    r ≤ (FormalMultilinearSeries.pi p).radius := by
+  apply le_of_forall_nnreal_lt (fun r' hr' ↦ ?_)
+  have I i : ∃ C > 0, ∀ n, ‖p i n‖ * (r' : ℝ) ^ n ≤ C :=
+    norm_mul_pow_le_of_lt_radius _ (hr'.trans_le (h i))
+  choose C C_pos hC using I
+  obtain ⟨D, D_nonneg, hD⟩ : ∃ D ≥ 0, ∀ i, C i ≤ D :=
+    ⟨∑ i, C i, Finset.sum_nonneg (fun i _ ↦ (C_pos i).le),
+      fun i ↦ Finset.single_le_sum (fun j _ ↦ (C_pos j).le) (Finset.mem_univ _)⟩
+  apply le_radius_of_bound _ D (fun n ↦ ?_)
+  rcases le_or_lt ((r' : ℝ)^n) 0 with hr' | hr'
+  · exact le_trans (mul_nonpos_of_nonneg_of_nonpos (by positivity) hr') D_nonneg
+  · simp only [pi]
+    rw [← le_div_iff₀ hr', ContinuousMultilinearMap.opNorm_pi,
+      pi_norm_le_iff_of_nonneg (by positivity)]
+    intro i
+    exact (le_div_iff₀ hr').2 ((hC i n).trans (hD i))
+
+lemma HasFPowerSeriesWithinOnBall.pi
+    (hf : ∀ i, HasFPowerSeriesWithinOnBall (f i) (p i) s e r) (hr : 0 < r) :
+    HasFPowerSeriesWithinOnBall (fun x ↦ (fun i ↦ f i x)) (FormalMultilinearSeries.pi p) s e r where
```

Comment:
Why not
```suggestion
    HasFPowerSeriesWithinOnBall (fun x i ↦ f i x) (FormalMultilinearSeries.pi p) s e r where
```
? (Many times)

Reply:
It's on purpose. Both are of course the same for a proficient Lean user, but for newcomers I think my spelling is more readable: you have a function of `x` which turns out to be a function. Since I'm really speaking about the power series structure for `x`, not `i`, I want to emphasize they have different roles.

Reply:
One more spelling: `fun x ↦ (f · x)`.